python_tests/artifact/test_backend.py

```diff
@@ -0,0 +1,100 @@
+from unittest.mock import MagicMock
+
+from optuna_dashboard.artifact import _backend
+import pytest
+
+
+def test_get_artifact_path() -> None:
+    study = MagicMock(_study_id=0)
+    trial = MagicMock(_trial_id=0, study=study)
+    assert _backend.get_artifact_path(trial=trial, artifact_id="id0") == "/artifacts/0/0/id0"
+
+
+def test_artifact_prefix() -> None:
+    actual = _backend._artifact_prefix(trial_id=0)
+    assert actual == "dashboard:artifacts:0:"
+
+
+@pytest.fixture()
+def init_storage_with_artifact_meta() -> MagicMock:
+    storage = MagicMock()
+
+    get_study_system_attrs = (
+        lambda study_id: {
+            "dashboard:artifacts:0:id0": '{"artifact_id": "id0", "filename": "foo.txt"}',
+            "dashboard:artifacts:0:id1": '{"artifact_id": "id1", "filename": "bar.txt"}',
+            "baz": "baz",
+        }
+        if study_id == 0
+        else {}
+    )
+    storage.get_study_system_attrs.side_effect = get_study_system_attrs
+
+    trial0_system_attrs = {
+        "artifacts:id2": '{"artifact_id": "id2", "filename": "baz.txt"}',
+    }
+    trial1_system_attrs = {
+        "artifacts:id3": '{"artifact_id": "id3", "filename": "qux.txt"}',
+    }
+    get_trial_system_attrs = (
+        lambda trial_id: trial0_system_attrs
+        if trial_id == 0
+        else trial1_system_attrs
+        if trial_id == 1
+        else {}
+    )
+    storage.get_trial_system_attrs.side_effect = get_trial_system_attrs
+
+    get_all_trials = (
+        lambda study_id: [
+            MagicMock(
+                _trial_id=0, study=MagicMock(_study_id=study_id), system_attrs=trial0_system_attrs
+            ),
+            MagicMock(
+                _trial_id=1, study=MagicMock(_study_id=study_id), system_attrs=trial1_system_attrs
+            ),
+        ]
+        if study_id == 0
+        else []
+    )
+    storage.get_all_trials.side_effect = get_all_trials
+
+    return storage
```

Comment:
Overusing `MagicMock` can decrease the readability and make tests more fragile, increasing maintenance costs. How about creating an actual `InMemoryStorage` object instead of using a Mock like the test case below?

https://github.com/optuna/optuna-dashboard/blob/f1259305b564884f1b14598185c2c569c639c0a3/python_tests/artifact/test_optuna_compatibility.py#L19-L53